Makefile

```diff
@@ -199,80 +184,17 @@ esp32:
-	$(MAKE) -j $(NUM_CPU) -C toolchains/$(ESP32_CHIP)/
-
-.PHONY: build/$(ESP32_CHIP)/lib/libtoit_vm.a  # Marked phony to force regeneration.
-build/$(ESP32_CHIP)/lib/libtoit_vm.a: build/$(ESP32_CHIP)/CMakeCache.txt build/$(ESP32_CHIP)/include/sdkconfig.h
-	(cd build/$(ESP32_CHIP) && ninja toit_vm)
-
-.PHONY: build/$(ESP32_CHIP)/system.snapshot  # Marked phony to force regeneration.
-build/$(ESP32_CHIP)/system.snapshot: $(ESP32_SYSTEM_ENTRY) tools
-	mkdir -p build/$(ESP32_CHIP)
-	$(TOITC_BIN) -w $@ $<
-
-.PHONY: build/$(ESP32_CHIP)/flash/program.snapshot  # Marked phony to force regeneration.
-build/$(ESP32_CHIP)/flash/program.snapshot: $(ESP32_ENTRY) tools
-	mkdir -p build/$(ESP32_CHIP)/flash
-	$(TOITC_BIN) -w $@ $<
-
-build/$(ESP32_CHIP)/CMakeCache.txt: check-esp32-env
-	mkdir -p build/$(ESP32_CHIP)
-	(cd build/$(ESP32_CHIP) && cmake ../../ -G Ninja -DTOITC=$(TOITC_BIN) -DTOITPKG=$(TOITPKG_BIN) -DCMAKE_BUILD_TYPE=$(BUILD_TYPE) -DCMAKE_TOOLCHAIN_FILE=../../toolchains/$(ESP32_CHIP)/$(ESP32_CHIP).cmake --no-warn-unused-cli)
-
-build/$(ESP32_CHIP)/include/sdkconfig.h:
-	mkdir -p build/$(ESP32_CHIP)
-	$(MAKE) -C toolchains/$(ESP32_CHIP) -s "$(CURDIR)"/$@
-
-
-# ESP32 VARIANTS FLASH
-ifdef ESP32_ENTRY
-flash: build/$(ESP32_CHIP)/flash/program.snapshot
-endif
+esp32-no-env: check-env check-esp32-env sdk
+	IDF_TARGET=$(ESP32_CHIP) idf.py -C toolchains/$(ESP32_CHIP) -B build/$(ESP32_CHIP) -p "$(ESP32_PORT)" build
 
-.PHONY: flash
-flash: check-env-flash sdk esp32
-	mkdir -p build/$(ESP32_CHIP)/flash
-	cp build/$(ESP32_CHIP)/firmware.envelope build/$(ESP32_CHIP)/flash/firmware.envelope
-ifdef ESP32_ENTRY
-	$(FIRMWARE_BIN) -e build/$(ESP32_CHIP)/flash/firmware.envelope \
-	    container install program build/$(ESP32_CHIP)/flash/program.snapshot
-endif
-ifdef ESP32_WIFI_SSID
-	$(FIRMWARE_BIN) -e build/$(ESP32_CHIP)/flash/firmware.envelope \
-	    property set wifi '{"wifi.ssid": "$(ESP32_WIFI_SSID)", "wifi.password": "$(ESP32_WIFI_PASSWORD)"}'
-endif
-	$(FIRMWARE_BIN) -e build/$(ESP32_CHIP)/flash/firmware.envelope \
-	    extract --bootloader.bin -o build/$(ESP32_CHIP)/flash/bootloader.bin
-	$(FIRMWARE_BIN) -e build/$(ESP32_CHIP)/flash/firmware.envelope \
-	    extract --partitions.bin -o build/$(ESP32_CHIP)/flash/partitions.bin
-	$(FIRMWARE_BIN) -e build/$(ESP32_CHIP)/flash/firmware.envelope \
-	    extract --firmware.bin -o build/$(ESP32_CHIP)/flash/firmware.bin
-	python $(IDF_PATH)/components/esptool_py/esptool/esptool.py --chip $(ESP32_CHIP) --port $(ESP32_PORT) --baud 921600 \
-	    --before default_reset --after hard_reset write_flash -z --flash_mode dio --flash_freq 40m --flash_size detect \
-		0x001000 build/$(ESP32_CHIP)/flash/bootloader.bin \
-		0x008000 build/$(ESP32_CHIP)/flash/partitions.bin \
-		0x010000 build/$(ESP32_CHIP)/flash/firmware.bin
-
-.PHONY: check-env-flash
-check-env-flash:
-ifndef ESP32_PORT
-	$(error ESP32_PORT is not set)
-endif
+# ESP32 MENU CONFIG
+.PHONY: menuconfig
+menuconfig: check-esp32-env
```

Comment:
Would be nice if `menuconfig` also sourced the default esp-idf if it doesn't see any compiler. (like the `esp32`).